### ISSUE TITLE
ansible-playbook tab completion to select .yml playbook files not working

### ISSUE THREAD

**Issue by karmicgeezer** [robbyrussell/oh-my-zsh#7107]
Tab completion to select a yml file after 'ansible-playbook' doesn't work, I've tried adding ansible and ansible-playbook to the plugins. I've also tried with both .yml and .yaml with no success.

Autocompletion works without oh_my_zsh but can't figure out where in oh_my_zsh tab completion for file selection after the ansible-playbook command is used.

**Comment by karmicgeezer** [comment]
That’s what I would have expected, regular tab completion  command ‘ansible-playbook’ then tab to choose file or start type beginning of the file name then tab. Just a warning bell. 

This has happened since installing oh_my_zsh many months ago.

Command/file completion works well using the same command with out oh_my_zsh and all other shells.

**Comment by mcornella** [comment]
I can't reproduce your problem. What do `echo $_comps[ansible-playbook]` and `which -a ansible-playbook` show?

**Comment by karmicgeezer** [comment]
echo $_comps[ansible-playbook] returns: _ansible

which -a ansible-playbook returns: /usr/local/bin/ansible-playbook

Many thanks for taking a look at this, greatly appreciated.

**Comment by jayooin** [comment]
Where did you get that _ansible plugin ? is it in your custom plugins ?

**Comment by karmicgeezer** [comment]
From https://github.com/zsh-users/zsh-completions

I've tried deleting the following from ~/.oh-my-zsh:
./custom/plugins/zsh-completions/src/_ansible*
./custom/plugins/ansible*

Removed any reference of ansible plugins from .zshrc with no success, tab just gives me a warning bell.

'brew uninstall --force zsh-completions' seems to have fixed the issue!! 

Thank you all so much for pointing me in the right direction, I'll play with getting zsh-completions installed with brew and removing the ansible specific completions.

Again, very grateful!!

**Comment by mcornella** [comment]
You should give zsh-completions a second chance, it looks like [they support `ansible-playbook`](https://github.com/zsh-users/zsh-completions/blob/master/src/_ansible-playbook), you just had it wrongly tied with the `_ansible` completion file. Perhaps you have `compdef _ansible ansible-playbook` somewhere in your zshrc configuration file?

**Comment by jasperf** [comment]
There does not seem to be ansible or ansible-vault support for zsh-completions. Another repo git https://github.com/clarketm/zsh-completions ~/.oh-my-zsh/plugins/zsh-completions does but it also does not do path completion after type `ansible-vault view ..` it does show command options.

```
➜  trellis git:(master) ✗ echo $_comps[ansible-playbook] 
_ansible
➜  trellis git:(master) ✗ echo $_comps[ansible-vault]    
_ansible
➜  trellis git:(master) ✗ which -a ansible-vault
/usr/local/bin/ansible-vault
```

**Comment by mike-code** [comment]
⬆️ I have the same problem as @jasperf with `ansible-vault`

**Comment by bartenbach** [comment]
I just have generic zsh with no plugins and can't tab complete anything from ansible-vault either.  Ansible-playbook works fine though.

**Comment by mcornella** [comment]
`ansible-vault` looks to [be working on zsh 5.8](https://github.com/zsh-users/zsh/commit/5ff3529caf8bc31725714eef19a10d7a93f8be87), you should try to update your zsh version.

**Comment by bartenbach** [comment]
Nice, thanks for the heads up. I can confirm this does work on my Mac with zsh 5.8. Unfortunately my main issue, though, was using it on CentOS at work. I'll check this out on Monday.

**Comment by robinzoncruzo** [comment]
Regarding `ansible-vault`, looks like it works, but in wrong order:
Zsh completes command like `ansible-vault --vault-id id@/passwd/path view /vault/path`, but this is wrong format of call, it should be `ansible-vault view --vault-id id@/passwd/path /vault/path`. It's bug or something.

**Comment by mcornella** [comment]
> Regarding `ansible-vault`, looks like it works, but in wrong order: Zsh completes command like `ansible-vault --vault-id id@/passwd/path view /vault/path`, but this is wrong format of call, it should be `ansible-vault view --vault-id id@/passwd/path /vault/path`. It's bug or something.

This is fixed in https://github.com/zsh-users/zsh/commit/781129b838a4fbb8cb9a70f7dc62dd26a4bb9ba9, which will be released with zsh 5.9.
